src/Aspire.Dashboard/Model/ResourceViewModel.cs

```diff
@@ -34,11 +35,33 @@ public sealed class ResourceViewModel
     public required ImmutableArray<VolumeViewModel> Volumes { get; init; }
     public required FrozenDictionary<string, ResourcePropertyViewModel> Properties { get; init; }
     public required ImmutableArray<CommandViewModel> Commands { get; init; }
+
     /// <summary>The health status of the resource. <see langword="null"/> indicates that health status is expected but not yet available.</summary>
-    public required HealthStatus? HealthStatus { get; init; }
-    public required ImmutableArray<HealthReportViewModel> HealthReports { get; init; }
+    public HealthStatus? HealthStatus { get; private set; }
+
+    public required ImmutableArray<HealthReportViewModel> HealthReports
+    {
+        get => _healthReports;
+        init
+        {
+            _healthReports = value;
+            HealthStatus = ComputeHealthStatus(value, KnownState);
+        }
+    }
+
     public KnownResourceState? KnownState { get; init; }
 
+    private static HealthStatus? ComputeHealthStatus(ImmutableArray<HealthReportViewModel> healthReports, KnownResourceState? state)
+    {
+        if (healthReports.Length == 0 && state is KnownResourceState.Running)
```

Comment:
If the resource isn't running then return null.

Reply:
AppHost will already be sending null

Reply:
What will be null? Do you mean there won't be any reports if not running? Or the reports will all have null health status if not running?